File: SqlLite/app/src/main/java/com/example/sqllite/MainActivity.java
package com.example.sqllite;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {

    RecyclerView listCars;
    Button button;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listCars = findViewById(R.id.list_cars);
        button = findViewById(R.id.go_to_add_car);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), SecondActivity.class);
                startActivity(intent);
            }
        });

        DataBaseHelper dataBaseHelper = new DataBaseHelper(this);
        listCars.setLayoutManager(new LinearLayoutManager(this));
        listCars.setHasFixedSize(true);
        RecycleAdapter adapter = new RecycleAdapter(this, dataBaseHelper.getCarsList());
        listCars.setAdapter(adapter);
    }
}

File: Room/app/src/main/java/com/example/room/MainActivity.kt
package com.example.room

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity



class MainActivity : AppCompatActivity() {


    lateinit var signIn : Button
    lateinit var signUp : Button
    lateinit var FirstName : EditText
    lateinit var LastName : EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        signIn = findViewById(R.id.signIn)
        signUp = findViewById(R.id.signUp)
        FirstName = findViewById(R.id.firstName)
        LastName = findViewById(R.id.lastName)

        val db = AppDatabase.getAppDataBase(applicationContext)
        val userDao = db?.userDao()
        val users: List<User> = userDao!!.getAll()


        val pref : SharedPreferences
        pref = getSharedPreferences("1",MODE_PRIVATE)

        if(pref.getString("login","").toString() != "" && pref.getString("password","").toString() != ""){
            intent = Intent(this, UserActivity::class.java);
            startActivity(intent)
        }

        signUp.setOnClickListener{

            intent = Intent(this, SignUpActivity::class.java);
            startActivity(intent)
        }

        signIn.setOnClickListener{

            val first = FirstName.text.toString()
            val last = LastName.text.toString()

            for(i in users)
            {
                if (first == i.firstName)
                {
                    if (last == i.lastName){
                        pref.edit().putString("login", first).putString("password",last).commit()
                        intent = Intent(this, UserActivity::class.java);
                        startActivity(intent)
                    }
                }
            }
        }

    }
}
